Explorer/Assets/DCL/InWorldCamera/PhotoDetail/PhotoDetailController.cs

```diff
@@ -0,0 +1,197 @@
+            }
+
+            if (waitForAnimation)
+                DelayedHideDeleteModalAsync().Forget();
+            else
+                HideDeleteModal();
+        }
+
+        private void JumpInClicked() =>
+            isClosing = true;
+
+        private void ToggleInfoSidePanel()
+        {
+            metadataSidePanelAnimator.ToggleSizeMode(toFullScreen: metadataPanelIsOpen, viewInstance.sidePanelAnimationDuration);
+            metadataPanelIsOpen = !metadataPanelIsOpen;
+        }
+
+        protected override void OnViewShow()
+        {
+            viewInstance!.infoButton.onClick.AddListener(ToggleInfoSidePanel);
+            viewInstance!.previousScreenshotButton.onClick.AddListener(ShowPreviousReel);
+            viewInstance!.nextScreenshotButton.onClick.AddListener(ShowNextReel);
+            viewInstance!.downloadButton.onClick.AddListener(DownloadReelClicked);
+            viewInstance!.linkButton.onClick.AddListener(CopyReelLinkClicked);
+            viewInstance!.twitterButton.onClick.AddListener(ShareReelClicked);
+            viewInstance!.deleteButton.onClick.AddListener(ShowDeleteModal);
+            viewInstance!.cancelDeleteIntentButton?.onClick.AddListener(() => DeletionModalCancelClick());
+            viewInstance!.cancelDeleteIntentBackgroundButton?.onClick.AddListener(() => DeletionModalCancelClick(false));
+            viewInstance!.deleteReelButton?.onClick.AddListener(DeleteScreenshot);
+
+            ShowReel(inputData.CurrentReelIndex);
+        }
+
+        private void DeleteScreenshot()
+        {
+            if (!inputData.UserOwnedReels) return;
+
+            inputData.ExecuteDeleteAction(currentReelIndex);
+            isClosing = true;
+        }
+
+        protected override void OnViewInstantiated()
+        {
+            metadataSidePanelAnimator = new MetadataSidePanelAnimator(viewInstance!.rootContainer, viewInstance.infoButtonImageRectTransform);
+        }
+
+        protected override void OnViewClose()
+        {
+            viewInstance!.infoButton.onClick.RemoveListener(ToggleInfoSidePanel);
+            viewInstance!.previousScreenshotButton.onClick.RemoveListener(ShowPreviousReel);
+            viewInstance!.nextScreenshotButton.onClick.RemoveListener(ShowNextReel);
+            viewInstance!.downloadButton.onClick.RemoveListener(DownloadReelClicked);
+            viewInstance!.linkButton.onClick.RemoveListener(CopyReelLinkClicked);
+            viewInstance!.twitterButton.onClick.RemoveListener(ShareReelClicked);
+            viewInstance!.deleteButton.onClick.RemoveListener(ShowDeleteModal);
+            HideDeleteModal();
+
+            viewInstance.mainImageCanvasGroup.alpha = 0;
+            photoDetailInfoController.Release();
+        }
+
+        protected override void OnBeforeViewShow()
+        {
+            currentReelIndex = inputData.CurrentReelIndex;
+            viewInstance!.deleteButton.gameObject.SetActive(inputData.UserOwnedReels);
```

Comment:
As in the previous comment, controllers should not modify how the view looks by touching its controls directly.

Reply:
same as before